lib/src/graph/utilities.dart

```diff
@@ -40,3 +43,13 @@ extension SetUtilities<T> on Set<T> {
     }
   }
 }
+
+extension ElementSource on Element {
```

Comment:
This is probably ok because this is a non-library with pretty limited scope, but in general it's good to be conservative with extension methods, particularly when there are inheritance hierarchies involved.  They can't be mocked for tests, and they can do some pretty non-intuitive things, and a change that affects the inferred type can change behavior. e.g. consider 

```
import 'dart:math';
class A {}
class B extends A {
 doThing() { print('world');}
}
A get a => b;
B b = B();
void main() {
   // a == b, but the calls are to different methods.
   a.doThing();
   b.doThing();
   var foo = Random().nextBool() ? a : b;
   // Always uses the extended
   foo.doThing();
   if (foo is B) {
     // Always prints, and always uses the regular method
     foo.doThing();
   }
}
extension Extension on A {
  doThing() {
    print('hello');
   }
}
```

Reply:
I was originally going to add another method (which is why I think I made this change in the first place), but since it is just the two existing methods I'll switch them back to local methods 👍